api/app/Http/Controllers/ApiController.php

```diff
@@ -0,0 +1,218 @@
+     * @param  Request  $request
+     * @return Response
+     */
+    public function putMany(Request $request)
+    {
+        $data = $request->data;
+        $idName = $this->repository->getModel()->getKeyName();
+        $ids = [];
+        foreach ($data as $piece) {
+            $ids[] = $piece[$idName];
+        }
+        $models = $this->repository->findMany($ids);
+
+        $putModels = [];
+        foreach ($data as $piece) {
+            $id = $piece[$idName];
+            if ($models->has($id)) {
+                $model = $models->get($id);
+            } else {
+                $model = $this->repository->getModel();
+            }
+            /** @var BaseModel $model */
+            $model->fill($piece);
+            $putModels[] = $model;
+        }
+
+        $models = $this->repository->saveMany($putModels);
+
+        $response = $this->responder->created();
+        $routes = [];
+        foreach ($models as $model) {
+            if ($route = RouteHelper::getRoute($model)) {
+                $routes[] = $route;
+            }
+        }
+        $response->setContent(json_encode($routes));
+
+        return $response;
+    }
+
+    /**
+     * Patch an entity.
+     *
+     * @param  string   $id
+     * @param  Request  $request
+     * @return Response
+     */
+    public function patchOne($id, Request $request)
+    {
+        $model = $this->repository->find($id);
+        $model->fill($request->all());
+        $this->repository->save($model);
+
+        return $this->responder->noContent();
+    }
+
+    /**
+     * Patch many entites.
+     *
+     * @param  Request  $request
+     * @return Response
+     */
+    public function patchMany(Request $request)
+    {
+        $data = $request->data;
+        $idName = $this->repository->getModel()->getKeyName();
+        $ids = [];
+        foreach ($data as $piece) {
+            $ids[] = $piece[$idName];
+        }
+        $models = $this->repository->findMany($ids);
+        if ($models->count() !== count($ids)) {
+            throw new \InvalidArgumentException('Not all entities were found');
+        }
+
+        foreach ($data as $piece) {
+            $id = $piece[$idName];
+            $model = $models->get($id);
+
+            /** @var BaseModel $model */
+            $model->fill($piece);
+        }
+
+        $this->repository->saveMany($models);
+
+        return $this->responder->noContent();
+    }
+
+    /**
+     * Delete an entity.
+     *
+     * @param  string   $id
+     * @return Response
+     */
+    public function deleteOne($id)
+    {
+        $model = $this->repository->find($id);
+        $this->repository->delete($model);
+
+        return $this->responder->noContent();
+    }
+
+    /**
+     * Delete many entites.
+     *
+     * @param  Request  $request
+     * @return Response
+     */
+    public function deleteMany(Request $request)
+    {
+        $ids =$request->data;
+        $models = $this->repository->findMany($ids);
```

Comment:
The `deleteMany` implementation should follow the same pattern as the `patchMany` and `putMany`. They will have the same request structures (a `DELETE` request probably won't include the rest of the entity.